src/Observers/ChangeHistoryObserver.php

```
@@ -0,0 +1,64 @@
+                    $newHistory->model()->associate($model);
+                    $newHistory->save();
+                }
+            }
+        }
+    }
+
+    public function deleting(Model $model): void
+    {
+        if (config('history.enabled', false) && method_exists($model, 'history')) {
+            /** @var MorphMany $history */
+            $history = $model->history();
```

PHPStan (2.1.31)

PHPDoc tag @var for variable $history contains generic class Illuminate\Database\Eloquent\Relations\MorphMany but does not specify its types: TRelatedModel, TDeclaringModel


(missingType.generics)